File: 9-data_structures_stacks+queues/stack_arr.js
class Stack {
  constructor() {
    this.array = [];
  }

  peek() {
    if (this.array.length === 0) {
      return null;
    }

    return this.array[this.array.length - 1];
  }

  push(value) {
    this.array.push(value);
    return this;
  }

  pop() {
    this.array.pop();
    return this;
  }
}

const s = new Stack();
s.push('google');
s.push('udemy');
s.push('discord');
console.log(s.peek());
s.pop();
s.pop();
s.pop();
console.log(s);

File: 9-data_structures_stacks+queues/stack_arr.rb
class Stack
  def initialize
    @arr = []
  end

  def peek
    @arr.last
  end

  def push(value)
    @arr << value
  end

  def pop
    @arr.pop
  end
end

s = Stack.new
s.push('google')
s.push('udemy')
s.push('discord')
puts s.peek
s.pop
s.pop
s.pop
p s
